src/TypeRegistry.php

```diff
@@ -244,6 +235,15 @@ public static function init() {
 			do_action( 'graphql_register_types' );
 		}
 
+		/**
+		 * Register all Union Types
+		 * Unions need to be registered after other types as they reference other Types
+		 */
+		require_once WPGRAPHQL_PLUGIN_DIR . 'src/Type/Union/CommentAuthorUnion.php';
+		require_once WPGRAPHQL_PLUGIN_DIR . 'src/Type/Union/MenuItemObjectUnion.php';
+		require_once WPGRAPHQL_PLUGIN_DIR . 'src/Type/Union/PostObjectUnion.php';
+		require_once WPGRAPHQL_PLUGIN_DIR . 'src/Type/Union/TermObjectUnion.php';
+
```

Comment:
So, realistically, Types _should_ be able to be registered in any order. For example, what if you have a Union that returns a Connection Type? Or a Connection that returns a Union Type? 

Types should be able to be registered in any order. Their fields _should_ be built as a callback, once all Types have been registered, thus allowing for circular Types. 

I think we might have some places where we're not fully respecting the concept of circular Types, and that might be what needs to be addressed more than moving things lower in the Registry.

Reply:
I agree with that principle however the Introspection query does not. If the one of the types in the Union is defined during the `graphql_register_type` action. The introspection query resolve that type to `null`. This is the error I receive when the `Product` post-type found in the MenuItemObjectUnion using on line [#22](https://github.com/wp-graphql/wp-graphql/blob/develop/src/Type/Union/MenuItemObjectUnion.php#L22)

```
Error: Introspection result missing possibleTypes: {"kind":"UNION","name":"MenuItemObjectUnion","description":null,"fields":null,"inputFields":null,"interfaces":null,"enumValues":null,"possibleTypes":null}
    at w (http://example.com/wordpress/wp-content/plugins/wp-graphiql/assets/app/build/static/js/main.b1825402.js?ver=5.2.3:1:586976)
    at b (http://example.com/wordpress/wp-content/plugins/wp-graphiql/assets/app/build/static/js/main.b1825402.js?ver=5.2.3:1:586263)
    at r (http://example.com/wordpress/wp-content/plugins/wp-graphiql/assets/app/build/static/js/main.b1825402.js?ver=5.2.3:1:585723)
    at http://example.com/wordpress/wp-content/plugins/wp-graphiql/assets/app/build/static/js/main.b1825402.js?ver=5.2.3:1:588833
    at Array.map (<anonymous>)
    at r (http://example.com/wordpress/wp-content/plugins/wp-graphiql/assets/app/build/static/js/main.b1825402.js?ver=5.2.3:1:588810)
    at http://example.com/wordpress/wp-content/plugins/wp-graphiql/assets/app/build/static/js/main.b1825402.js?ver=5.2.3:1:564667
```

Reply:
Ok, ya, we've got a different error to address here then. I think I may have addressed it in my upcoming PR that allows the Schema to be cleared.